### PR TITLE
gambit-scheme, gerbil-scheme: build fixes

### PR DESCRIPTION
**Pull request by eraserhd**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### Gambit

`gambit-scheme` has an incorrectly hard-coded path for OpenSSL on Mac.  This has been fixed upstream, but they will not make a release for "about a month," so this patches `config.status`.  

This made it so that, even though Gambit would build, nothing could be built with it - it would use the 1.1 OpenSSL headers and link against 1.0, which has missing symbols.

### Gerbil

1. Gerbil attempted to find Gambit's `gsi` in the path, but it is not added to the path.  I patched the script to use the full path to `gsi`.
2. Gerbil attempts to find it's home directory from argv0 and got confused by Homebrew's symlinks, so `GERBIL_HOME` is now defaulted to `#{libexec}`.

### CODE REVIEW

**Comment by eraserhd** [comment]
Build failed on `brew style`.  I fixed the style issue and pushed a commit to note that contributors shoudl run `brew style foo` in CONTRIBUTING.md.

**Comment by eraserhd** [comment]
@ober

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ brew uninstall --force foo
 brew install --build-from-source foo
 brew test foo
 brew audit --strict foo
+brew style foo
```
Comment (SMillerDev) [inline CONTRIBUTING.md]: Please make this a separate pull request.

Comment (eraserhd) [reply CONTRIBUTING.md]: done

**Comment by chenrui333** [comment]
Thanks @eraserhd!
